lib/lightning_web/live/run_live/workorder_component.ex

```diff
@@ -17,39 +17,68 @@ defmodule LightningWeb.RunLive.WorkOrderComponent do
      socket
      |> assign(assigns)
      |> assign(project: project, can_rerun_job: can_rerun_job)
-     |> set_work_order_details(work_order)}
+     |> set_details(work_order)}
   end
 
   def update(%{work_order: work_order} = assigns, socket) do
-    {:ok, socket |> assign(assigns) |> set_work_order_details(work_order)}
+    {:ok, socket |> assign(assigns) |> set_details(work_order)}
   end
 
   def update(assigns, socket) do
     {:ok, assign(socket, assigns)}
   end
 
-  defp set_work_order_details(socket, work_order) do
-    last_run = List.last(List.first(work_order.attempts).runs)
+  defp set_details(socket, work_order) do
+    last_run = get_last_run(work_order)
+    last_run_finished_at = format_finished_at(last_run)
+    work_order_inserted_at = Calendar.strftime(work_order.inserted_at, "%c %Z")
+    workflow_name = work_order.workflow.name || "Untitled"
 
-    last_run_finished_at =
-      case last_run do
-        %{finished_at: %_{} = finished_at} ->
-          Calendar.strftime(finished_at, "%c %Z")
+    assign_details(
+      socket,
+      work_order,
+      last_run,
+      last_run_finished_at,
+      work_order_inserted_at,
+      workflow_name
+    )
+  end
 
-        _ ->
-          nil
-      end
+  defp get_last_run(work_order) do
+    work_order.attempts
+    |> List.first()
+    |> case do
+      nil -> nil
+      attempt -> List.last(attempt.runs)
+    end
+  end
 
-    work_order_inserted_at = Calendar.strftime(work_order.inserted_at, "%c %Z")
+  defp format_finished_at(last_run) do
+    case last_run do
+      %{finished_at: %_{} = finished_at} ->
+        Calendar.strftime(finished_at, "%c %Z")
+
+      _ ->
+        nil
+    end
+  end
 
+  defp assign_details(
```

Comment:
A nit pick suggest, I think we can remove this indirection as it only assigns the fields without any other additional info or processing.

Reply:
Thanks for the catch @jyeshe. I did the change